File: backend/app/schemas/parent.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, EmailStr

from app.schemas.child import ChildResponse


class ParentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    organization_id: uuid.UUID
    display_name: str
    email: EmailStr | None
    phone: str | None
    created_at: datetime
    updated_at: datetime


class ParentDetailResponse(ParentResponse):
    children: list[ChildResponse]


class ParentChildCreate(BaseModel):
    child_id: uuid.UUID

File: frontend-admin/src/types/parent.ts
import type { Child } from './child'

export type Parent = {
  id: string
  organization_id: string
  display_name: string
  email: string | null
  phone: string | null
  created_at: string
  updated_at: string
}

export type ParentDetail = Parent & { children: Child[] }
